x/ecocredit/server/core/bridge_receive_test.go

```diff
@@ -0,0 +1,316 @@
+		Batch: &core.MsgBridgeReceive_Batch{
+			Recipient: recipient,
+			Amount:    "3",
+			OriginTx: &core.OriginTx{
+				Id:     "0x1324092835908235",
+				Source: "polygon:0x325325230958",
+			},
+			StartDate: &start,
+			EndDate:   &end,
+			Metadata:  "",
+			Note:      "bridged from test",
+		},
+		Project: &core.MsgBridgeReceive_Project{
+			ReferenceId:  projectRefId,
+			Jurisdiction: "US-KY",
+			Metadata:     "hi",
+			ClassId:      "C01",
+		},
+	}
+	res, err := s.k.BridgeReceive(s.ctx, &msg)
+	assert.NilError(t, err)
+	assert.Equal(t, res.ProjectId, project.Id)
+	assert.Equal(t, res.BatchDenom, batch.Denom)
+
+	// this was a fresh account, so we know their balance is only what was bridged to it.
+	bal, err := s.k.Balance(s.ctx, &core.QueryBalanceRequest{
+		Address:    recipient,
+		BatchDenom: batch.Denom,
+	})
+	assert.NilError(t, err)
+	balAfter := bal.Balance
+	assert.Equal(t, balAfter.TradableAmount, "3")
+}
+
+func TestBridgeReceive_ProjectNoBatch(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	recipient := testutil.GenAddress()
+	refId := "VCS-001"
+	project, batch := setupBridgeTest(s, refId)
+	startDate, endDate := (&timestamppb.Timestamp{Seconds: 500}).AsTime(), (&timestamppb.Timestamp{Seconds: 600}).AsTime()
+
+	msg := core.MsgBridgeReceive{
+		Issuer: s.addr.String(),
+		Batch: &core.MsgBridgeReceive_Batch{
+			Recipient: recipient,
+			Amount:    "3",
+			OriginTx: &core.OriginTx{
+				Id:     "0x12345",
+				Source: "polygon:0x12345",
+			},
+			StartDate: &startDate,
+			EndDate:   &endDate,
+			Metadata:  "hi",
+			Note:      "bridged test",
+		},
+		Project: &core.MsgBridgeReceive_Project{
+			ReferenceId:  refId,
+			Jurisdiction: "US-KY",
+			ClassId:      "C01",
+		},
+	}
+
+	res, err := s.k.BridgeReceive(s.ctx, &msg)
+	assert.NilError(t, err)
+	assert.Equal(t, res.ProjectId, project.Id)
+	assert.Check(t, res.BatchDenom != batch.Denom)
+
+	batch, err = s.stateStore.BatchTable().GetByDenom(s.ctx, res.BatchDenom)
+	assert.NilError(t, err)
+
+	bal, err := s.k.Balance(s.ctx, &core.QueryBalanceRequest{
+		Address:    recipient,
+		BatchDenom: batch.Denom,
+	})
+	assert.NilError(t, err)
+	assert.Equal(t, bal.Balance.TradableAmount, msg.Batch.Amount)
+}
+
+func TestBridgeReceive_None(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+	setupBridgeTest(s, "VCS-002")
+	recipient := testutil.GenAddress()
+	start, end := time.Now(), time.Now()
+	msg := core.MsgBridgeReceive{
+		Issuer: s.addr.String(),
+		Batch: &core.MsgBridgeReceive_Batch{
+			Recipient: recipient,
+			Amount:    "3",
+			OriginTx: &core.OriginTx{
+				Id:     "0x12345",
+				Source: "polygon:0x12345",
+			},
+			StartDate: &start,
+			EndDate:   &end,
+			Metadata:  "bar",
+			Note:      "bridged",
+		},
+		Project: &core.MsgBridgeReceive_Project{
+			ReferenceId:  "VCS-001",
+			Jurisdiction: "US-KY",
+			Metadata:     "foo",
+			ClassId:      "C01",
+		},
+	}
+	res, err := s.k.BridgeReceive(s.ctx, &msg)
+	assert.NilError(t, err)
+
```

Comment:
Just to be safe, could we explicitly assert that the project bridged to is not `VCS-002` ?

Reply:
done in https://github.com/regen-network/regen-ledger/pull/1174/commits/8842f9b99f1aaa5347a30946ec929a48e7d84d1d